runtime/onert/backend/cpu/ops/SplitLayer.cc

```diff
@@ -34,37 +34,23 @@ SplitLayer::SplitLayer() : _input(nullptr), _num_splits(0), _axis(0), _outputs()
   // DO NOTHING
 }
 
-void SplitLayer::splitFloat32()
+template <typename T> void SplitLayer::splitGeneric(void)
 {
   nnfw::cker::SplitParams op_params;
   op_params.axis = _axis;
   op_params.num_split = _num_splits;
 
-  std::vector<nnfw::cker::Shape *> outputDimsPtr;
-  std::vector<nnfw::cker::Shape> outputDims;
-  outputDimsPtr.reserve(_num_splits);
-  outputDims.reserve(_num_splits);
-
-  for (uint32_t i = 0; i < _num_splits; i++)
-  {
-    outputDims.push_back(getTensorShape(_outputs[i]));
-    outputDimsPtr.push_back(&outputDims[i]);
-  }
-
-  std::vector<float *> outputFloatPtrs;
+  std::vector<T *> outputPtrs;
 
   for (const auto output : _outputs)
   {
-    outputFloatPtrs.emplace_back(reinterpret_cast<float *>(output->buffer()));
+    outputPtrs.emplace_back(reinterpret_cast<T *>(output->buffer()));
   }
 
-  nnfw::cker::Split<float>(op_params, getTensorShape(_input),
-                           reinterpret_cast<float *>(_input->buffer()), getTensorShape(_outputs[0]),
-                           outputFloatPtrs.data());
+  nnfw::cker::Split<T>(op_params, getTensorShape(_input), reinterpret_cast<T *>(_input->buffer()),
```

Comment:
I think, getTensorShape(_outputs[0]) paramter will NOT have the correct values if passed like this. This is used in split kernel assuming this has the axis populated. Not sure if this has when called here.

Reply:
As far as I checked, the shape inferencing rules ensure that each tensor within `_outputs` have the required shape and dimension information populated. Please let me know if anything is missed.